To: vim_dev@googlegroups.com
Subject: Patch 7.3.356
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.356
Problem:    Using "o" with 'cindent' set may freeze Vim. (lolilolicon)
Solution:   Skip over {} correctly. (Hari G)
Files:	    src/misc1.c


*** ../vim-7.3.355/src/misc1.c	2011-10-04 18:03:43.000000000 +0200
--- src/misc1.c	2011-11-30 12:56:55.000000000 +0100
***************
*** 6127,6133 ****
  
  /*
   * Set w_cursor.col to the column number of the last unmatched ')' or '{' in
!  * line "l".
   */
      static int
  find_last_paren(l, start, end)
--- 6127,6133 ----
  
  /*
   * Set w_cursor.col to the column number of the last unmatched ')' or '{' in
!  * line "l".  "l" must point to the start of the line.
   */
      static int
  find_last_paren(l, start, end)
***************
*** 6140,6146 ****
  
      curwin->w_cursor.col = 0;		    /* default is start of line */
  
!     for (i = 0; l[i]; i++)
      {
  	i = (int)(cin_skipcomment(l + i) - l); /* ignore parens in comments */
  	i = (int)(skip_string(l + i) - l);    /* ignore parens in quotes */
--- 6140,6146 ----
  
      curwin->w_cursor.col = 0;		    /* default is start of line */
  
!     for (i = 0; l[i] != NUL; i++)
      {
  	i = (int)(cin_skipcomment(l + i) - l); /* ignore parens in comments */
  	i = (int)(skip_string(l + i) - l);    /* ignore parens in quotes */
***************
*** 7953,7958 ****
--- 7953,7959 ----
  			 * If we're at the end of a block, skip to the start of
  			 * that block.
  			 */
+ 			l = ml_get_curline();
  			if (find_last_paren(l, '{', '}')
  				&& (trypos = find_start_brace(ind_maxcomment))
  							    != NULL) /* XXX */
*** ../vim-7.3.355/src/version.c	2011-11-30 11:31:25.000000000 +0100
--- src/version.c	2011-11-30 13:02:52.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     356,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
213. Your kids start referring to you as "that guy in front of the monitor."

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
